trie/trienode/node.go

```diff
@@ -78,7 +78,7 @@ func NewNodeSet(owner common.Hash) *NodeSet {
 // ForEachWithOrder iterates the nodes with the order from bottom to top,
 // right to left, nodes with the longest path will be iterated first.
 func (set *NodeSet) ForEachWithOrder(callback func(path string, n *Node)) {
-	var paths []string
+	paths := make([]string, 0, len(set.Nodes))
```

Comment:
It could be useful. As normally we have thousands of dirty nodes in each block, could be beneficial in hash-based scheme.